doc/en/user/source/production/config.rst

```diff
@@ -219,6 +219,15 @@ GeoServer provides a number of facilities to control external entity resolution:
   
   This setting prevents ``ENTITY_RESOLUTION_ALLOWLIST`` from being used.
 
+Spring Security Firewall
```

Comment:
Good to have information here, but we should have a mention in the upgrade notes as well, e.g.:

https://docs.geoserver.org/stable/en/user/installation/upgrade.html#notes-on-upgrading-specific-versions